### ISSUE TITLE
Custom evaluation metric module

### ISSUE THREAD

**Issue by RemyLau** [pyg-team/pytorch_geometric#3349]
I'm wondering would it be possible to implement a module for **custom evaluation metrics**. There are a few reasons why this is desirable. The most obvious reason is **performance**. Currently, all the evaluations are from ``scikit-learn``, rendering _extremely_ low GPU usage (I can provide measurements if needed). In the past, I've written custom metric functions in a more engineered way to massively speed up the computation.

On a separate but related note, would it be possible to have the option to disable evaluation logging during training? Specifically, disabling this block. https://github.com/pyg-team/pytorch_geometric/blob/ec952b48e2a80136a253eb1e5399cb55cac91e67/torch_geometric/graphgym/train.py#L23-L27

Having this option would further increase the GPU usage by disabling unnecessary transfer to the CPU (correct me if I'm wrong). Also, the ``ogbn-proteins`` example script has this desired behavior of being able to choose the evaluation logging step. https://github.com/snap-stanford/ogb/blob/9d7a3f5e89640938f93b52388cfe259ae8fe3855/examples/nodeproppred/proteins/gnn.py#L160

**Comment by JiaxuanYou** [comment]
Thanks for your suggestion. In my daily use cases, this logger update takes negligible time. Could you describe your use case, where this could be a bottleneck?
Also, some metrics, such as AUC ROC, it not straightforward to implement on GPU, while `scikit-learn` provides convenient support.
I'm happy to hear your suggestions. Also, PR for more evaluation metrics is welcomed!
